assets/src/components/map.tsx

```diff
@@ -329,34 +381,38 @@ const Map = (props: Props): ReactElement<HTMLDivElement> => {
   return (
     <>
       <div className={`m-vehicle-map-state ${autoCenteringClass}`} />
-      <ReactLeafletMap
+      <MapContainer
         className="m-vehicle-map"
         id="id-vehicle-map"
-        ref={mapRef}
         maxBounds={[
           [41.2, -72],
           [43, -69.8],
         ]}
         zoomControl={false}
         center={defaultCenter}
         zoom={13}
-        onmovestart={() => {
-          // If the user drags or zooms, they want manual control of the map.
-          // But don't disable shouldAutoCenter if the move was triggered by an auto center.
-          if (!isAutoCentering.current) {
-            setShouldAutoCenter(false)
-          }
-        }}
-        onmoveend={() => {
-          // Wait until the auto centering is finished to start listening for manual moves again.
-          if (isAutoCentering.current) {
-            isAutoCentering.current = false
-          }
+        // workaround for ref unsupported in react-leaflet v3.
+        // can be replaced with directly passing ref in v4.
+        whenCreated={(mapInstance) => {
+          mapRef.current = mapInstance
         }}
       >
+        <EventAdder
+          isAutoCentering={isAutoCentering}
+          setShouldAutoCenter={setShouldAutoCenter}
+        />
+        <Autocenterer
+          shouldAutoCenter={shouldAutoCenter}
+          isAutoCentering={isAutoCentering}
+          latLngs={latLngs}
+        />
         <ZoomControl position="topright" />
         <FullscreenControl position="topright" />
-        <RecenterControl turnOnAutoCenter={() => setShouldAutoCenter(true)} />
+        <RecenterControlButton
+          position="topright"
+          recenter={() => setShouldAutoCenter(true)}
+        />
+
         <TileLayer
           url={`${tilesetUrl()}/{z}/{x}/{y}.png`}
           attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
```

Comment:
It looks like at one point we explicitly set the attribution (probably just to make it shorter and take up less space?), but now it displays the default attribution with the Leaflet link (on both prod and dev-green). If it's possible, it would be nice to fix that as part of this upgrade.

Reply:
Setting `prefix={false}` in [8fc146a](https://github.com/mbta/skate/pull/1739/commits/8fc146a1c5fc42eaccc3cada0f59136eee32a31e) did the trick